generate_rospkg_apkbuild/APKBUILD.em.sh

```diff
@@ -124,12 +145,44 @@ check() {
     --catkin-make-args run_tests 2>&1 | tee $checklog
   catkin_test_results 2>&1 | tee $checklog
 @[  end if]@
-@[  if use_cmake]@
-  cd src/$_pkgname/build
+@[  if (use_ament_cmake or use_ament_python) and ros2_workspace_available]@
+  source /usr/ros/@(ros_distro)/setup.sh
+@[  end if]@
+@[  if use_ament_cmake or use_ament_python]@
+  export PYTHONPATH="$builddir"/tmp/pkg/usr/ros/@(ros_distro)/lib/python${PYTHON_VERSION}/site-packages:${PYTHONPATH}
+  export AMENT_PREFIX_PATH="$builddir"/tmp/pkg/usr/ros/@(ros_distro):${AMENT_PREFIX_PATH}
+  export PATH="$builddir"/tmp/pkg/usr/ros/@(ros_distro)/bin:${PATH}
+  mkdir -p "$builddir"/tmp/pkg
+@[  end if]@
+@[  if use_cmake or use_ament_cmake]@
+  cd build
+@[  if use_ament_cmake]@
+  make install DESTDIR="$builddir"/tmp/pkg
+@[  end if]@
   if [ $(make -q test > /dev/null 2> /dev/null; echo $?) -eq 1 ]; then
     make test 2>&1 | tee $checklog
   fi
 @[  end if]@
+@[  if use_ament_python]@
+  cd src/$_pkgname
+  python setup.py \
+    build \
+    --build-base="$builddir"/tmp/build \
+    install \
+    --root="$builddir"/tmp/pkg \
+    --prefix=/usr/ros/@(ros_distro) 2>&1 | tee $buildlog
+  TEST_TARGET=$(ls -d */ | grep -m1 "\(test\|tests\)") || true
+  if [ -z "$TEST_TARGET" ]; then
+    echo "No \"test\" or \"tests\" directory. Check skipped" | tee $checklog
+    return 0
+  fi
+  USE_PYTEST=$(grep '\<pytest\>' setup.py) || true
```

Comment:
[optional] I'm not sure if it is better but we could use something like this to determine if there are any `pytest` test:
```
pytest --collect-only -qq | head -n -2 | wc -l
```